src/mam4xx/water_uptake.hpp

```diff
@@ -58,7 +59,11 @@ class Water_Uptake {
 };
 
 namespace water_uptake {
-
+//  number of variables in state_q
+constexpr int nvars = 40;
+constexpr int maxd_aspectype = 14;
```

Comment:
is this the number of distinct interstitial + cloudeborne species indices?

Reply:
According to comments in the F90 code, it is the number of  vapor and aerosol number/mass mixing ratio variables in state%q [#/kg or kg/kg].